### ISSUE TITLE
Warn when using `%` operator on an f-string

### ISSUE THREAD

**Issue by RenzoMXD** [astral-sh/ruff#24159]
## Summary

Currently, none of the F50x rules fire when the left-hand side of `%` is an f-string, because the check only matches `Expr::StringLiteral`. This means code like:

```python
f"{banana}" % banana
f"hello %s %s" % (1, 2)
```

produces no warnings, even though mixing f-string interpolation with `%`-formatting is almost certainly a mistake — both serve the same purpose.

## Proposal

Add a new rule (RUF072) that flags any use of `%` on an f-string. I can't think of a valid use case where using `%` over f-string interpolation is desired.

## Related

- #24031 — original report
- #24142 — F507 fix for non-tuple RHS

**Comment by ai-mountain** [comment]
@MichaReiser Can I have your opinions on this new rule suggestion?

**Comment by ai-mountain** [comment]
@MichaReiser  please review my PR

**Comment by ntBre** [comment]
Thanks for your contributions! However, please give us some time to respond to issues and PRs before pinging us. We're subscribed to all activity on the repo, so we've gotten notifications for the issue being opened, your comments, and also for the PR. It can be a bit overwhelming to manage all of those along with the other notifications coming in.

**Comment by MichaReiser** [comment]
Thanks. I think this makes sense
